### PR TITLE
cmd: Fix mismatched types in testModuleData

### PR DESCRIPTION
**Pull request by Amulyam24**
Rectify the values of `testModuleData` with the correct
types in `TestCCCheckCLiFunction` in kata-check_(!x86)_test.go

Fixes: #2735

Signed-off-by: Amulya Meka <amulmek1@in.ibm.com>

### CODE REVIEW

File: src/runtime/cmd/kata-runtime/kata-check_ppc64le_test.go
Comment (Jakob-Naucke) [inline src/runtime/cmd/kata-runtime/kata-check_ppc64le_test.go:51]: Shouldn't that be `false`?

File: src/runtime/cmd/kata-runtime/kata-check_s390x_test.go
Comment (Jakob-Naucke) [inline src/runtime/cmd/kata-runtime/kata-check_s390x_test.go:50]: Same as above

**Review by Jakob-Naucke — COMMENTED** [review]
LGTM otherwise. This is also good to know because I thought unit tests _were_ running on the !x86 CI jobs.

File: src/runtime/cmd/kata-runtime/kata-check_ppc64le_test.go
Comment (Amulyam24) [reply src/runtime/cmd/kata-runtime/kata-check_ppc64le_test.go:51]: @Jakob-Naucke , the boolean value is for representing if it is a directory and it is one right? `/sys/module/kvm`

**Review by Jakob-Naucke — APPROVED** [review]
> @Jakob-Naucke , the boolean value is for representing if it is a directory and it is one right? `/sys/module/kvm/`

yeah... probably just wrong all along. Thank you!

**Comment by Amulyam24** [comment]
> LGTM otherwise. This is also good to know because I thought unit tests _were_ running on the !x86 CI jobs.

@Jakob-Naucke , what are your thoughts on having a basic test on !x86 to run the unit tests as a part of the CI jobs?

**Comment by Jakob-Naucke** [comment]
> @Jakob-Naucke , what are your thoughts on having a basic test on !x86 to run the unit tests as a part of the CI jobs?

So I could have sworn there was something in the jobs that went "on non-x86, do all the static checks too, because only for x86 are they done on GitHub Actions", but I can't find it any more and I might be wrong. [This](https://github.com/kata-containers/tests/blob/6838b47bf4f82e6c70c7f032e40865bea01f6888/.ci/run.sh#L137-L143) exists but isn't run in common jobs.

**Comment by Amulyam24** [comment]
> > @Jakob-Naucke , what are your thoughts on having a basic test on !x86 to run the unit tests as a part of the CI jobs?
> 
> So I could have sworn there was something in the jobs that went "on non-x86, do all the static checks too, because only for x86 are they done on GitHub Actions", but I can't find it any more and I might be wrong. 

okay, I'm not sure if I've come across one such test either.

> [This](https://github.com/kata-containers/tests/blob/6838b47bf4f82e6c70c7f032e40865bea01f6888/.ci/run.sh#L137-L143) exists but isn't run in common jobs.

AFAIU, the [make test](https://github.com/kata-containers/tests/blob/6838b47bf4f82e6c70c7f032e40865bea01f6888/Makefile#L166) runs the default CI JOB which here is kubernetes test suite.

**Review by cmaf — APPROVED** [review]

**Comment by Amulyam24** [comment]
> no, [there it is](https://github.com/kata-containers/tests/blob/6838b47bf4f82e6c70c7f032e40865bea01f6888/.ci/jenkins_job_build.sh#L130-L141)
> 
Thanks @Jakob-Naucke!
> > We also run static checks on travis for x86 and ppc64le
> 
> ouch. Todo:
> 
> * Find why the test from this PR wasn't run.
> * Enable for non-x86.

Yes, I'll look into it and get back on this in another issue(if required).
> 
> e: btw, I was so sure it was there because I would have done something like that earlier and found it was already there.

**Comment by Amulyam24** [comment]
@Jakob-Naucke , I agree, the sooner the better, to unblock the CI.

> why did this not hit us in [kata-containers/tests#4027](https://github.com/kata-containers/tests/pull/4027)?

Since the static checks are run against the [kata_repo](https://github.com/kata-containers/tests/blob/acddd90cdf6981a71a4b90a73b8d8bc4cf09b088/.ci/jenkins_job_build.sh#L138) passed here, which is `tests` repo for tests#4207.

**Comment by Jakob-Naucke** [comment]
It seems to be fixed, so I'm giving it one last shot before merging with a red CLH. /retest-clh-k8s-containerd

**Comment by Jakob-Naucke** [comment]
> > why did this not hit us in [kata-containers/tests#4027](https://github.com/kata-containers/tests/pull/4027)?
> 
> Since the static checks are run against the [kata_repo](https://github.com/kata-containers/tests/blob/acddd90cdf6981a71a4b90a73b8d8bc4cf09b088/.ci/jenkins_job_build.sh#L138) passed here, which is `tests` repo for tests#4207.

Makes sense. Would you know why [this](http://jenkins.katacontainers.io/job/kata-containers-2.0-ubuntu-s390x-PR/694) is coming up though even though this PR passed?

**Comment by Amulyam24** [comment]
> > > why did this not hit us in [kata-containers/tests#4027](https://github.com/kata-containers/tests/pull/4027)?
> > 
> > 
> > Since the static checks are run against the [kata_repo](https://github.com/kata-containers/tests/blob/acddd90cdf6981a71a4b90a73b8d8bc4cf09b088/.ci/jenkins_job_build.sh#L138) passed here, which is `tests` repo for tests#4207.
> 
> Makes sense. Would you know why [this](http://jenkins.katacontainers.io/job/kata-containers-2.0-ubuntu-s390x-PR/694) is coming up though even though this PR passed?

@Jakob-Naucke , I'm seeing this for the first time. Didn't come across this error on `ppc64le` when I tested for static checks. I'm not quite sure on what's causing this. Probably you can open an issue to get some suggestions on it?

Also, one thing to be noted is that, the CI(s390x& ARM) for this PR were run before the static-checks PR was merged, see here - http://jenkins.katacontainers.io/job/kata-containers-2.0-ubuntu-s390x-PR/669/consoleFull. Hence seeing the error now.

**Comment by Jakob-Naucke** [comment]
Ah, a good old #3974.

**Pull request merged**
